galaxy/api/v2/views/collection.py

```diff
@@ -65,22 +66,19 @@ def post(self, request, *args, **kwargs):
 
         artifact = self._save_artifact(artifact_data)
 
-        import_task = models.ImportTask.objects.create(
-            owner=request.user,
-            state=models.ImportTask.STATE_PENDING,
-        )
-
-        async_result = enqueue_with_reservation(
-            tasks.import_collection, [],
-            kwargs={
-                'artifact_pk': artifact.pk,
-                'repository_pk': repository.pk,
-                'namespace_pk': namespace.pk,
-                'task_id': import_task.id,
-                'filename': data['filename'],
+        task = tasking.create_task(
+            tasks.import_collection,
+            task_cls=models.CollectionImport,
+            params={
+                'artifact_id': artifact.pk,
+                'repository_id': repository.pk,
+            },
+            task_args={
+                'namespace': namespace,
```

Comment:
Are you planning on adding API endpoints to query imports by namespace as part of this PR, or is that coming later?

Reply:
Yes, working on this in terms of this PR.

Reply:
Done